tests/test_config.py

```diff
@@ -0,0 +1,217 @@
+        ]
+    },
+    "test_global_local": {
+        "test_work_dir": "simple",
+        "expected_config": {
+            "user": "user@test.edulinq.org"
+        },
+        "expected_source": {
+            "user": (
+                "<default config file>:: %s"
+                % os.path.join('TEMP_DIR', 'simple', 'autograder.json')
+            )
+        },
+        "config_global": os.path.join("global", "autograder.json")
+    },
+    "test_cli_global": {
+        "test_work_dir": "empty-dir",
+        "expected_config": {
+            "user": "user@test.edulinq.org"
+        },
+        "expected_source": {
+            "user": (
+                "<cli config file>:: %s"
+                % os.path.join('TEMP_DIR', 'simple', 'autograder.json')
+            )
+        },
+        "config_paths": [os.path.join("simple", "autograder.json")],
+        "config_global": os.path.join("global", "autograder.json")
+    },
+    "test_local_cli": {
+        "test_work_dir": "simple",
+        "expected_config": {
+            "user": "user@test.edulinq.org"
+        },
+        "expected_source": {
+            "user": (
+                "<cli config file>:: %s"
+                % os.path.join('TEMP_DIR', 'old-name', 'config.json')
+            )
+        },
+        "config_paths": [os.path.join("old-name", "config.json")],
+    },
+    "test_local_cli_global": {
+        "test_work_dir": "simple",
+        "expected_config": {
+            "user": "user@test.edulinq.org"
+        },
+        "expected_source": {
+            "user": (
+                "<cli config file>:: %s"
+                % os.path.join('TEMP_DIR', 'old-name', 'config.json')
+            )
+        },
+        "config_paths": [os.path.join("old-name", "config.json")],
+        "config_global": os.path.join("global", "autograder.json")
+    }
+}
+
+class TestConfig(unittest.TestCase):
+    def test_all(self):
+        for test in TESTS:
+            self._evaluate_test_config(**TESTS.get(test))
+
+    def _evaluate_test_config(
+            self, test_work_dir, expected_config,
+            expected_source, config_global = "",
+            config_paths = None):
+        """
+        Prepares testing environment while normalizes cli config paths,
+        global config path and expected source paths. Evaluates the given expected and
+        source configs with actual get_tiered_config() output.
+        """
+
+        previous_work_directory = os.getcwd()
+
+        temp_dir = autograder.util.dir.get_temp_dir(prefix = 'autograder-test-config-')
+        global_config = os.path.join(temp_dir, config_global)
+
+        abs_config_paths = []
+        if config_paths is not None:
+            for rel_config_path in config_paths:
+                abs_config_paths.append(os.path.join(temp_dir, rel_config_path))
+
+        autograder.util.dirent.copy_contents(CONFIGS_DIR, temp_dir)
+
+        cwd = os.path.join(temp_dir, test_work_dir)
```

Comment:
I don't like the name "cwd", since it is not always accurate throughout the function.
I would use a name symmetric with "previous_work_directory".